### ISSUE TITLE
Some Accessibility issues

### ISSUE THREAD

**Issue by zhou-zhoukang** [maksimowiczm/find-my-ip#11]
Hi! Thank you for your hard work on this great open-source app. I notice a few accessibility issues. 

Taking the example of the interface shown in the figure below, the red boxes indicate the order in which components are focused sequentially by screen readers when visually impaired users navigate through the UI. When they focus on the Switch, they will only hear its current on/off state but cannot understand what specific feature this Switch is enabling/disabling. This requires them to perform additional left/right swipes to obtain contextual information about the Switch. This may creates unnecessary cognitive load.

<img src="https://github.com/user-attachments/assets/5e1227b3-7d3f-44cc-b9ce-c9834eedfacb" width=400/>

Perhaps a more accessible approach in this case would be to set the `onCheckedChange` of the Switch to null, and instead handle the click event in the `ListItem`. This is just a suggestion based on what I’ve observed, and I hope it may be helpful.
```
ListItem(
    modifier = Modifier.clickable { onCheckedChange() },
    trailingContent = {
        Switch(
            onCheckedChange = null,
        )
    }
)
```

Alternatively, adding an `onClickLabel` to the Switch might also be a good practice. An example of how this can be implemented is provided in the code snippet below. For instance, when considering a Switch, a screen reader might announce "Double tap to **activate**" before a label is added. After adding an appropriate label, it would instead say "Double tap to **Open/Close VPN**", which can make the switch's function more intuitive and clear to users. 

```
Switch(
    modifiler = Modifier.semantics {  
       onClick(label = "Open VPN") { true }
    }
)
```

Please feel absolutely no pressure — I completely understand if this takes a back seat to other priorities. Truly appreciate all the effort and hard work you've already put in!

**Comment by maksimowiczm** [comment]
Thank you for the detailed explanation. I wasn’t aware that notification screen accessibility had such limited support. I’ll work on improving it.

I appreciate you pointing this out; I haven’t used the TalkBack feature before, so I wasn’t fully aware of the importance of proper semantic setup.

Also, how do you display those red boxes? That looks like a useful tool.

**Comment by maksimowiczm** [comment]
I made some accessibility improvements in the latest version. Please let me know if you have any further comments.

**Comment by zhou-zhoukang** [comment]
Thanks a lot for addressing the accessibility improvements. I checked the updates in version 4.0.1 and they look fantastic! This will definitely enhance the experience for all users. Appreciate your dedication and responsiveness!
